lib/calendar.ts

```diff
@@ -29,19 +29,33 @@ import type {
   AnyTemporalKey,
   CalendarSlot
 } from './internaltypes';
+import type { CalendarFieldDescriptor } from './ecmascript';
 
 const ArrayIncludes = Array.prototype.includes;
 const ArrayPrototypePush = Array.prototype.push;
+const ArrayPrototypeSort = Array.prototype.sort;
 const IntlDateTimeFormat = globalThis.Intl.DateTimeFormat;
-const ArraySort = Array.prototype.sort;
 const MathAbs = Math.abs;
 const MathFloor = Math.floor;
 const ObjectCreate = Object.create;
 const ObjectEntries = Object.entries;
+const OriginalMap = Map;
 const OriginalSet = Set;
+const OriginalWeakMap = WeakMap;
 const ReflectOwnKeys = Reflect.ownKeys;
+const MapPrototypeEntries = Map.prototype.entries;
+const MapPrototypeGet = Map.prototype.get;
+const MapPrototypeSet = Map.prototype.set;
 const SetPrototypeAdd = Set.prototype.add;
-const SetPrototypeValues = Set.prototype.values;
+const WeakMapPrototypeGet = WeakMap.prototype.get;
+const WeakMapPrototypeSet = WeakMap.prototype.set;
+
+const MapIterator = ES.Call(MapPrototypeEntries, new Map(), []);
```

Comment:
We can, but I was trying to keep divergence from upstream to a minimum until we get all the big rebases done. 

Although this commit (https://github.com/js-temporal/temporal-polyfill/pull/275/commits/72cb533e9825c397cbc9da366201b67cf32389e9) seems to be exclusively caching built-ins and not changing any logic at all. Maybe we should just drop that upstream commit entirely?

@ptomato @12wrigja what do you think?

If we did drop it, then there'd be more rebase conflicts with upstream, but I don't expect calendar.ts to change a lot going fwd so this may not be too bad.

Reply:
OK, I'll drop both this commit and the other one that's exclusively primordial-focused (and other future commits that are focused only on primordials) while otherwise accepting primordial-caching changes that are embedded in commits that also change behavior outside primordial caching.

Reply:
Actually, I liked some of the changes in https://github.com/js-temporal/temporal-polyfill/commit/72cb533e9825c397cbc9da366201b67cf32389e9, so I kept that commit and just removed some of the more confusing changes like replacing iterators and caching WeakMap primordials. See [fixup commit](https://github.com/js-temporal/temporal-polyfill/pull/275/commits/2dbce3bbec0fba18ea404ae9f410c6ce1f349edc). Resolving.